packages/region/src/services/region-module.ts

```diff
@@ -223,37 +178,171 @@ export default class RegionModuleService<
     data?: UpdatableRegionFields,
     @MedusaContext() sharedContext: Context = {}
   ): Promise<Region[]> {
-    let result: Region[] = []
+    let normalizedInput: UpdateRegionDTO[] = []
     if (isString(idOrSelectorOrData)) {
-      result = await this.regionService_.update(
-        [{ id: idOrSelectorOrData, ...data }],
-        sharedContext
-      )
+      normalizedInput = [{ id: idOrSelectorOrData, ...data }]
     }
 
     if (Array.isArray(idOrSelectorOrData)) {
-      result = await this.regionService_.update(
-        idOrSelectorOrData,
-        sharedContext
-      )
+      normalizedInput = idOrSelectorOrData
     }
 
     if (isObject(idOrSelectorOrData)) {
-      let toUpdate: Partial<UpdateRegionDTO>[] = []
       const regions = await this.regionService_.list(
-        { ...idOrSelectorOrData },
+        idOrSelectorOrData,
         {},
         sharedContext
       )
 
-      regions.forEach((region) => {
-        toUpdate.push({ id: region.id, ...data })
+      normalizedInput = regions.map((region) => ({
+        id: region.id,
+        ...data,
+      }))
+    }
+    normalizedInput = RegionModuleService.normalizeInput(normalizedInput)
+
+    // If countries are being updated for a region, first make previously set countries' region to null to get to a clean slate.
+    // Somewhat less efficient, but region operations will be very rare, so it is better to go with a simple solution
+    const regionsWithCountryUpdate = normalizedInput
+      .filter((region) => !!region.countries)
+      .map((region) => region.id)
+      .flat()
+
+    if (regionsWithCountryUpdate.length > 0) {
```

Comment:
nit: checking > 0 is not necessary, 0 will be considered as false and otherwise true. But I am fine keeping it

Reply:
No I think it's fine without it, I'm just so used to typing it in golang 😄